examples/multi-threaded/main.c

```diff
@@ -61,14 +63,18 @@ static void pcb(struct mg_connection *c, int ev, void *ev_data, void *fn_data) {
 static void fn(struct mg_connection *c, int ev, void *ev_data, void *fn_data) {
   if (ev == MG_EV_HTTP_MSG) {
     struct mg_http_message *hm = (struct mg_http_message *) ev_data;
+      char param[64];
+      mg_http_get_var(&hm->query, "param", param, sizeof(param));
+      if (param[0] == '\0') strcpy(param,"hi");
     if (mg_http_match_uri(hm, "/fast")) {
       // Single-threaded code path, for performance comparison
       // The /fast URI responds immediately
-      mg_http_reply(c, 200, "Host: foo.com\r\n", "hi\n");
+      mg_http_reply(c, 200, "Host: foo.com\r\n", param);
     } else {
       // Multithreading code path
       int sock = mg_mkpipe(c->mgr, pcb, c, true);             // Create pipe
       start_thread(thread_function, (void *) (size_t) sock);  // Start thread
+      mg_send(c->fn_data, param, strlen(param));
```

Comment:
I'm not sure this is correct. c->fn_data is NULL, I think.
Do you intend to pass parameters to a thread function?
Then instead of passing a socket to `start_thread()`, we should alloc a struct, copy socket and parameter there, and pass a pointer to a struct as a thread function arg.

Reply:
Agree with those points. However I still think we should not do it this way.

Reasoning:
1. The anticipated use case of multi-threaded functionality is a single GET/POST request with some parameters, and a thread function performs long calculations
2. The situation when a client streams extra data , IMO, is significantly more complex, rare, and probably should be handled via websocket anyway. 
3. We should cover (1) in our example
4. The question asked in a discussion, IMO, also talks about situation (1). A thread function just needs to know some request parameters - and I think passing those as a message body is the way to go, it would be just an ordinary REST request which requires a long calculation
5. The examples' pipe event handler is already complex to understand, cause it interconnects a parent and pipe connections. Sending stuff via the c->fn_data is not obvious at all. A far more understood way is what I suggested, IMO: just  calloc thread params, and pass them to a thread function. That would be understood by anyone, and it would be easy to adopt - just change/add your stuff to the thread param struct. Yes it is an extra calloc, but it is simple and obvious.

So I still would argue to avoid using a shady send to c->fn_data.

Reply:
Understood, thanks for the explanation, will do that.